src/librustc_trans/trans/consts.rs

```diff
@@ -130,13 +132,14 @@ fn addr_of_mut(ccx: &CrateContext,
 
 pub fn addr_of(ccx: &CrateContext,
                cv: ValueRef,
+               align: machine::llalign,
                kind: &str)
                -> ValueRef {
     match ccx.const_globals().borrow().get(&cv) {
         Some(&gv) => return gv,
```

Comment:
Oh, right, it looks them up by value not by name. Sorry for being dense here. I'll fix that to upgrade the alignment as required and add a test for it.